### PR TITLE
feat: 25695 - refactor reconnect syncronizers

### LINKED ISSUE #25695: Refactor LearnerTreeView and TeacherTreeView
`LearningSynchronizer` and `TeachingSynchronizer` should start all required tasks, not views.
`TeacherTreeView` can be replaced with `RecordAccessor`  - all information is available here during reconnect.

### PR DESCRIPTION
**Pull request by akugal**
Refactor Learner/Teacher views by moving task creation to synchronizers.

No need for `TeacherView` since `RecordAccessor` can be used.
`LearnerView` is now responsible for providing what to send and handling responses, so it is renamed to `LearnerTreeExchanger`.
Synchronizer's `synchronize()` method now accepts parameters to do synchronization and becomes a single entry point for the consensus reconnect module to do virtual map synchronization. 

Resolves: #25695

### CODE REVIEW

File: platform-sdk/consensus-reconnect-impl/src/main/java/org/hiero/consensus/reconnect/impl/ReconnectStateTeacher.java [hunk 1]
```diff
@@ -99,16 +100,16 @@ public ReconnectStateTeacher(
...
         signingWeight = signedState.getSigningWeight();
         roster = signedState.getRoster();
         final VirtualMapState virtualMapState = signedState.getState();
         hash = virtualMapState.getHash();
-        final ReconnectConfig reconnectConfig = configuration.getConfigData(ReconnectConfig.class);
-        // The teacher view will be closed by TeacherSynchronizer in reconnect() below
-        teacherView = virtualMapState.getRoot().buildTeacherView(reconnectConfig);
+        teacherMap = virtualMapState.getRoot();
+
+        synchronizer =
```
Comment (thenswan) [inline platform-sdk/consensus-reconnect-impl/src/main/java/org/hiero/consensus/reconnect/impl/ReconnectStateTeacher.java]: Before this PR, the flow was:

1. `ReconnectStatePeerProtocol` got `teacherState`.
2. It built `ReconnectStateTeacher`.
3. Inside the `ReconnectStateTeacher` constructor, the code immediately created a detached teacher view from the map.
4. Only after that, `ReconnectStatePeerProtocol` closed `teacherState`.
5. This was safe because the teacher already had its own detached snapshot.

After this PR, the flow changed:

1. `ReconnectStatePeerProtocol` gets `teacherState`.
2. It builds `ReconnectStateTeacher`.
3. The constructor now only stores `teacherMap = virtualMapState.getRoot()`.
4. `ReconnectStatePeerProtocol` then closes `teacherState`.
5. Later, when `teacher.execute()` runs, `TeachingSynchronizer.synchronize()` calls `teacherMap.detach()`.

The problem is step 5. By then, the reservation has already been closed. So the code may be trying to detach or read from a state that is no longer safely reserved.

File: platform-sdk/swirlds-virtualmap/src/main/java/com/swirlds/virtualmap/sync/LearningSynchronizer.java [hunk 2]
```diff
@@ -26,100 +46,141 @@ public class LearningSynchronizer {
...
-        out.start();
-
-        InterruptedException interruptException = null;
         try {
-            view.startLearnerTasks(workGroup, in, out);
+            // Perform the root-node (path 0) request/response handshake synchronously before forking
+            // any parallel tasks. The root response carries the teacher's first/last leaf path range,
+            // which must be known before the traversal order can be started and before any parallel
+            // send tasks can generate meaningful non-root requests.
+            exchangeRootNode(exchanger, input, output);
```
Comment (thenswan) [inline platform-sdk/swirlds-virtualmap/src/main/java/com/swirlds/virtualmap/sync/LearningSynchronizer.java:144]: The problem is: if that root exchange fails, the new code does this:

1. catches the exception,
2. calls `exchanger.abortOnException()`,
3. throws a new exception.

But it does **not** tell the `StandardWorkGroup` about the error.

That matters because `StandardWorkGroup.handleError()` is what runs the `breakConnection` callback and shuts down/interupts the worker threads. In production, `breakConnection` is usually `connection::disconnect`.

Before this PR, the learner view caught root-handshake errors and called `workGroup.handleError(e)` before rethrowing. That meant:

1. the connection got disconnected,
2. async stream threads were interrupted,
3. the other side stopped waiting.

After this PR, on an early root exchange failure, the learner may return/throw while those async stream threads are still alive, and the teacher may keep waiting until a socket timeout.

File: platform-sdk/swirlds-virtualmap/src/main/java/com/swirlds/virtualmap/sync/TeachingSynchronizer.java [hunk 3]
```diff
@@ -137,4 +175,48 @@ protected AsyncOutputStream buildOutputStream(
...
+     * response containing the teacher's first/last leaf path range. This must complete before
+     * any parallel tasks are forked so the learner can initialize its traversal order and begin
+     * sending non-root requests.
+     *
+     * @param teacherView teacher view used to access information about state
+     * @param in          the async input stream to read the root request from
+     * @param out         the async output stream to send the root response to
+     * @throws MerkleSynchronizationException if the exchange fails, times out, or is interrupted
+     */
+    private void exchangeRootNode(RecordAccessor teacherView, final AsyncInputStream in, final AsyncOutputStream out) {
```
Comment (thenswan) [inline platform-sdk/swirlds-virtualmap/src/main/java/com/swirlds/virtualmap/sync/TeachingSynchronizer.java]: nit: nonnull, final

File: platform-sdk/swirlds-virtualmap/src/main/java/com/swirlds/virtualmap/sync/LearningSynchronizer.java [hunk 4]
```diff
@@ -26,100 +46,141 @@ public class LearningSynchronizer {
...
-            }
+        if (workGroup.hasExceptions()) {
+            exchanger.abortOnException();
             throw new MerkleSynchronizationException(
                     "Synchronization failed with exceptions", firstReconnectException.get());
         } else {
-            view.onSuccessfulComplete();
+            logger.info(RECONNECT.getMarker(), "learner is done synchronizing");
+            logger.info(RECONNECT.getMarker(), reconnectStats::format);
+            return exchanger.onSuccessfulComplete();
```
Comment (thenswan) [inline platform-sdk/swirlds-virtualmap/src/main/java/com/swirlds/virtualmap/sync/LearningSynchronizer.java]: Before this PR, `ReconnectStateLearner` owned `vmapLearner` and called `vmapLearner.abortOnException()` for any exception from `synchronizer.synchronize()`. Now `vmapLearner` is hidden behind `LearnerTreeExchanger`, but the success path calls `exchanger.onSuccessfulComplete()` outside the cleanup `try/catch`.

That method calls `VirtualMapLearner.finish()`, which is not just a getter: it waits for leaf deletion, waits for hashing, flushes reconnect data, and creates the final `VirtualMap`. If any of that throws, the exception escapes without calling `exchanger.abortOnException()`, so reconnect resources may not be cleaned up.

**Review by thenswan — DISMISSED** [review]

**Review by thenswan — APPROVED** [review]

**Review by poulok — APPROVED** [review]
